Step2

jougene opened the pull request:
Correctness of function naming (PSR)- with caliing linter for one file.

jougene committed:
new branch for step 2

jougene committed:
delete token from travis yaml

jougene committed:
add camel case test

jougene committed:
work with php parser

jougene committed:
update TODO

jougene committed:
гзвфеу ЕЩВЩ

Commit:
update TODO - second step

Commit:
update TODO - add description for phpparser

Commit:
add basic check functions' name

Commit:
update readme

Commit:
update readme

Commit:
add right collect functions names

Commit:
add NodeVisitor

Commit:
update TODO

jougene committed:
update TODO

add done marks to finished points

Commit:
test comments

Commit:
test comments

Commit:
test comments

Commit:
Merge branch 'step2' of https://github.com/jougene/hexlet-psr-linter into step2

Conflicts:
	README.md

Commit:
test badges

Commit:
add colors to output

Commit:
add linenumbers to output

now outout looks like:
Line 23: Method incorrect_name should be declared in CamelCase

mokevnin commented on src/Linter.php:
```diff
@@ -3,21 +3,50 @@
+        return $this->errors;
+    }
+
+    public static function isCamelCase(string $string) : bool
```
Прочитай про god object. Этой функции тут не место.

mokevnin commented on src/Linter.php:
```diff
@@ -3,21 +3,50 @@
+            $nodes = $parser->parse($code);
+            $nodes = $traverser->traverse($nodes);
+        } catch (Error $e) {
+            echo 'Parse Error: ', $e->getMessage();
```
Побочный эффект

Commit:
Add check with array of errors

Commit:
delete debug info

Commit:
Add Checkers file

Commit:
add Rules, Checkers

add rules for expanding application by new rules;
add checkers, that will be check code with the rules

pyldin601 commented on src/Checkers.php/FunctionNaming.php:
```diff
@@ -0,0 +1,7 @@
+<?php
+namespace HexletPsrLinter\Checkers;
+
+function checkFunctionsNaming(Array $array)
```
`array` пишется с маленькой буквы. это тип, а не класс.

pyldin601 commented on src/Linter.php:
```diff
@@ -3,21 +3,46 @@
 class Linter
 {
-    public function lint(string $code) : bool
+    private $errors = [];
```
Зачем линтеру хранить ошибки?

pyldin601 commented on src/Linter.php:
```diff
@@ -3,21 +3,46 @@
+            echo 'Parse Error: ', $e->getMessage();
+        }
+
+        return $nodeVisitor->getFunctions();
```
Зачем тут возвращаются функции?

jougene committed:
add

jougene committed:
Merge branch 'step2' of github.com:jougene/hexlet-psr-linter into step2

jougene committed:
refactoring structure of project
